File: Backend/src/main/java/com/dxc/dxc_platform/service/ProjectService.java
package com.dxc.dxc_platform.service;

import com.dxc.dxc_platform.dto.ProjectDto;
import com.dxc.dxc_platform.enums.ProjectStatus;
import com.dxc.dxc_platform.dto.ManagerProjectItemDto;
import com.dxc.dxc_platform.dto.ManagerProjectReviewDto;
import com.dxc.dxc_platform.dto.UserDto;
import java.util.List;

public interface ProjectService {

    ProjectDto createProject(ProjectDto request);

    ProjectDto updateProject(Long projectId, ProjectDto request);

    ProjectDto getProjectById(Long projectId);
    void updateProjectStatus(Long id, ProjectStatus status);

    List<ProjectDto> getAllProjects(String query, ProjectStatus status);

    List<ProjectDto> getMyProjects(String query, ProjectStatus status);

    ProjectDto setDeletedStatus(Long projectId, boolean deleted);

    List<ManagerProjectItemDto> getManagerProjects();

    List<UserDto.Summary> getChefsProjetForSelect();

    ProjectDto reviewProjectByManager(ManagerProjectReviewDto request);
    ProjectDto assignTeamToProject(Long projectId, Long teamId);
    List<ProjectDto> getMyAssignedProjects(String query, ProjectStatus status);
}

File: Frontend/src/app/core/services/project.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams, HttpResponse } from '@angular/common/http';
import { Observable } from 'rxjs';

export type ProjectStatus =
  | 'PRE_VALIDE'
  | 'EN_COURS'
  | 'EN_VALIDATION'
  | 'VALIDE'
  | 'REJETE'
  | 'CLOTURE';

export type RiskLevel = 'FAIBLE' | 'MOYEN' | 'ELEVE';

export interface ProjectDto {
  id?: number;
  name: string;
  description?: string;
  client: string;
  progressPercentage: number;
  status?: ProjectStatus;
  riskLevel: RiskLevel;
  startDate: string;
  endDate: string;
  teamId?: number;
  teamName?: string;
  deleted?: boolean;
  createdAt?: string;
  updatedAt?: string;
  managerId?: number;
  managerName?: string;

  managerComment?: string;
  reviewedAt?: string;
   chefProjetId?: number;
  chefProjetName?: string;
}

@Injectable({
  providedIn: 'root'
})
export class ProjectService {
  private readonly API_URL = 'http://localhost:8080/api/projects';

  constructor(private http: HttpClient) {}

  createProject(project: ProjectDto): Observable<HttpResponse<any>> {
    return this.http.post<any>(this.API_URL, project, {
      observe: 'response'
    });
  }

  updateProject(projectId: number, project: ProjectDto): Observable<HttpResponse<any>> {
    return this.http.put<any>(`${this.API_URL}/${projectId}`, project, {
      observe: 'response'
    });
  }

  getProjectById(projectId: number): Observable<ProjectDto> {
    return this.http.get<ProjectDto>(`${this.API_URL}/${projectId}`);
  }

  getAllProjects(query?: string, status?: string): Observable<ProjectDto[]> {
    let params = new HttpParams();

    if (query && query.trim()) {
      params = params.set('query', query.trim());
    }

    if (status && status.trim()) {
      params = params.set('status', status.trim());
    }

    return this.http.get<ProjectDto[]>(this.API_URL, { params });
  }

  getMyProjects(query?: string, status?: string): Observable<ProjectDto[]> {
    let params = new HttpParams();

    if (query && query.trim()) {
      params = params.set('query', query.trim());
    }

    if (status && status.trim()) {
      params = params.set('status', status.trim());
    }

    return this.http.get<ProjectDto[]>(`${this.API_URL}/my-projects`, { params });
  }

  setDeletedStatus(projectId: number, deleted: boolean): Observable<ProjectDto> {
    const params = new HttpParams().set('deleted', deleted.toString());
    return this.http.patch<ProjectDto>(`${this.API_URL}/${projectId}/deleted`, null, { params });
  }
  assignTeamToProject(projectId: number, teamId: number): Observable<ProjectDto> {
  return this.http.patch<ProjectDto>(`${this.API_URL}/${projectId}/assign-team?teamId=${teamId}`, {});
}
  
}